tests/unittests/CoreText/CTFontManagerTests.mm

```diff
@@ -85,7 +85,83 @@
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(kCTFontManagerErrorInvalidFontData, CFErrorGetCode(error));
+
+    EXPECT_FALSE(CTFontManagerUnregisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(kCTFontManagerErrorInvalidFontData, CFErrorGetCode(error));
+}
+
+TEST(CTFontManager, ShouldFailToRegisterSameFontTwice) {
+    woc::unique_cf<CFStringRef> fontName{ CFSTR("WinObjC") };
+    NSURL* testFileURL = __GetURLFromPathRelativeToModuleDirectory(@"/data/WinObjC.ttf");
+    CFErrorRef error = nullptr;
+    EXPECT_TRUE(CTFontManagerRegisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> font{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, font);
+
+    StrongId<NSString> familyName = (NSString*)CTFontCopyFamilyName(font.get());
+    EXPECT_OBJCEQ(@"WinObjC", familyName);
+
+    EXPECT_FALSE(CTFontManagerRegisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_NE(nullptr, error);
+    EXPECT_EQ(kCTFontManagerErrorAlreadyRegistered, CFErrorGetCode(error));
+
+    EXPECT_TRUE(CTFontManagerUnregisterFontsForURL((__bridge CFURLRef)testFileURL, kCTFontManagerScopeSession, &error));
+    EXPECT_EQ(nullptr, error);
+
+    woc::unique_cf<CTFontRef> failedFont{ CTFontCreateWithName(fontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, failedFont);
+
+    familyName = (NSString*)CTFontCopyFullName(failedFont.get());
+    EXPECT_OBJCEQ(@"Segoe UI", familyName);
+}
+
+TEST(CTFontManager, ShouldRegisterMultipleFonts) {
+    woc::unique_cf<CFStringRef> firstFontName{ CFSTR("WinObjC") };
+    woc::unique_cf<CFStringRef> secondFontName{ CFSTR("WinObjC-Italic") };
+    NSArray* urls = @[
+        __GetURLFromPathRelativeToModuleDirectory(@"/data/WinObjC.ttf"),
+        __GetURLFromPathRelativeToModuleDirectory(@"/data/WinObjC-Italic.ttf")
+    ];
+    CFArrayRef errors = nullptr;
+    EXPECT_TRUE(CTFontManagerRegisterFontsForURLs((__bridge CFArrayRef)urls, kCTFontManagerScopeSession, &errors));
+    EXPECT_NE(nullptr, errors);
+    EXPECT_EQ(nullptr, CFArrayGetValueAtIndex(errors, 0));
+    EXPECT_EQ(nullptr, CFArrayGetValueAtIndex(errors, 1));
+    CFRelease(errors);
+
+    woc::unique_cf<CTFontRef> firstFont{ CTFontCreateWithName(firstFontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, firstFont);
+    StrongId<NSString> firstName = (NSString*)CTFontCopyFullName(firstFont.get());
+    EXPECT_OBJCEQ(@"WinObjC", firstName);
+
+    woc::unique_cf<CTFontRef> secondFont{ CTFontCreateWithName(secondFontName.get(), 20, nullptr) };
+    EXPECT_NE(nullptr, secondFont);
+    StrongId<NSString> secondName = (NSString*)CTFontCopyFullName(secondFont.get());
+    EXPECT_OBJCEQ(@"WinObjC Italic", secondName);
```

Comment:
you have repeated code here and in few other tests that do the same thing.
Consider using a general function. This can help add more tests in the future.